Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -40,8 +42,31 @@ namespace xr
     // Permission request ID used to uniquely identify our request in the callback when calling requestPermissions.
     const int PERMISSION_REQUEST_ID{ 8435 };
 
+    struct XrContextARCore : public IXrContextARCore {
```

Comment:
Any reason this needs to be a separate object and not an interface directly off of the `Impl`?

Reply:
no reason, can try that out as well

Reply:
looking back on this, we want this XrContextARCore to exist independent of a xr session. we could have the System::Impl extend this interface, but I don't think we can have the session::impl implement this interface. Probably still worth decreasing the number of structs, so ill update the system impl to implement IXrContextARCore

Reply:
actually, System::Impl already has an is initialized function that does not have the same usage as the IXrContextARCore::IsInitialized function. I think we should keep this XrContextARCore as a separate struct